src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -5445,6 +5445,209 @@ static void F2(object? w)
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(14, 26));
         }
 
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable()
+        {
+            var source = @"
+#nullable enable annotations
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable_01()
+        {
+            var source = @"
+#nullable enable annotations
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable_02()
+        {
+            var source = @"
+#nullable enable
+#nullable disable warnings
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35748, "https://github.com/dotnet/roslyn/issues/35748")]
+        public void Directive_Annotations_Enable_03()
+        {
+            var source = @"
+#nullable enable
+#nullable restore warnings
+class Program
+{
+    static void M(string? s)
+    {
+        _ = s.ToString();
```

Comment:
>_ = s.ToString(); [](start = 8, length = 17)

May be worth testing a variant of this test with project-level `/nullable:warnings`

Reply:
See `Directive_Warnings_Enable_04`

Reply:
Sorry, I meant literally the same code (including `#nullable enable` and `#nullable restore warnings`) but with `/nullable:warnings`.

---
In reply to: [294999973](https://github.com/dotnet/roslyn/pull/36464#discussion_r294999973) [](ancestors = 294999973)

Reply:
Thanks for pointing out this scenario, I'll add a few simple tests to exercise this ('restore' brings us back to non-default project-level setting).

---
In reply to: [295039031](https://github.com/dotnet/roslyn/pull/36464#discussion_r295039031) [](ancestors = 295039031,294999973)